ui/pages/confirmations/hooks/useSignatureEventFragment.ts

```diff
@@ -0,0 +1,34 @@
+import { useCallback } from 'react';
+import { generateSignatureUniqueId } from '../../../helpers/utils/metrics';
+import { updateEventFragment } from '../../../store/actions';
+import { useConfirmContext } from '../context/confirm';
+import { SignatureRequestType } from '../types/confirm';
+import { isSignatureTransactionType } from '../utils';
+
+import type { MetaMetricsEventFragment } from '../../../../shared/constants/metametrics';
+
+/**
+ * When a signature has been requested, there should be an event fragment created for it in
+ * {@see {@link app/scripts/lib/createRPCMethodTrackingMiddleware.js}.
+ * This hook method is used to update an existing signature event fragment for a signature confirmation.
+ */
+export const useSignatureEventFragment = () => {
+  const { currentConfirmation } = useConfirmContext() as {
```

Comment:
Just for audit sake, this assumes nothing outside of confirmations will ever want to add properties to the signature, nor will we want to update any properties on anything but the current signature.

Harmless for now given our use case, but worth keeping in mind.